pkgs/servers/home-assistant/build-custom-component/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib
+, home-assistant
+, makeSetupHook
+}:
+
+{ pname
+, version
+, format ? "other"
+, src
+, ...
+}@args:
+
+with home-assistant.python.pkgs; let
+  manifestRequirementsCheckHook = import ./manifest-requirements-check-hook.nix {
+    inherit makeSetupHook;
+    inherit (home-assistant) python;
+  };
+in buildPythonPackage ({
+  inherit pname version format src;
```

Comment:
Could you also inherit `patches`?

Reply:
This is a bit of a red-herring. I see now that patches should be included with the set update on line 36... I do not know why, but passing an array of patches to buildHomeAssistantComponent does not seem to apply the patches (this previously worked when I passed them to stdenvNoCC.mkDerivation and symlinked the result to /var/lib/hass/custom_components/${X}).

Reply:
Patching seems to work fine for me.

```diff
diff --git a/pkgs/servers/home-assistant/custom-components/prometheus-sensor/default.nix b/pkgs/servers/home-assistant/custom-components/prometheus-sensor/default.nix
index 07bcd9abec1c..298828ca825e 100644
--- a/pkgs/servers/home-assistant/custom-components/prometheus-sensor/default.nix
+++ b/pkgs/servers/home-assistant/custom-components/prometheus-sensor/default.nix
@@ -14,6 +14,10 @@ buildHomeAssistantComponent rec {
     hash = "sha256-10COLFXvmpm8ONLyx5c0yiQdtuP0SC2NKq/ZYHro9II=";
   };
 
+  patches = [
+    ./test.patch
+  ];
+
   dontBuild = true;
 
   meta = with lib; {
```

leads to

```
prometheus-sensor> patching sources
prometheus-sensor> applying patch /nix/store/fvh4hp8iglcr1hlf6qqixb6s21n1sn5p-test.patch
prometheus-sensor> patching file test.md
```